Search/LinearSearch.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * @function linearSearch
+ * @description linear search is the simplest search possible in a array 
+ *  it has a linear cost, if the value is present in the array, then the index of the first occurence will be returned 
+ *  if it's not present, the return it will be -1 
+ * @param {number[]} array - list of numbers
+ * @param {number} target - target number to search for
+ * @return {number} - index of the target number in the list, or -1 if not found
+ * @see https://en.wikipedia.org/wiki/Linear_search\
+ * @example linearSearch([1,2,3,5], 3) => 2
+ * @example linearSearch([1,5,6], 2) => -1
+ */
+export const linearSearch = (array: any[], target: any): number => {
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] == target) return i;
+  }
+  return -1;
```

Comment:
Consider making this return the insertion index if the element is not found

Reply:
Ah well, linear search can also be written for sorted arrays. If this linear search is to be used with unsorted arrays as well, that's fine too (hence the "consider").